### DISCUSSION TITLE
Simulation of water intrusion with pressure drop

### DISCUSSION THREAD

**Discussion by livashu** [PMEAL/OpenPNM#2393] [Q&A]
I would like to set inlet and outlet pressure and simulate water ingress according to capillary pressure considering pressure drop, is this possible by ordinary percolation?

**Comment by jgostick** [comment]
Can you clarify a few things? 
1. Is water the wetting phase, so it's actually imbibition rather than drainage?
2. Is the water rate fast, such that viscous effects may be important?
3. If you apply a constant pressure, does that mean that the flow rate changes with time?
4. Do you need the pore-by-pore progress of the ingress, or is just the final equilibrium configuration sufficient?

**Reply by livashu** [reply]
1. Yes
2. I assume so.
3. Yes
4. It would be appreciated to see the progress of the infiltration, but just the final equilibrium state would be fine.

**Reply by jgostick** [reply]
1. OpenPNM can do imbibition with the MixedInvasionPercolation class.  See [this example](https://github.com/PMEAL/OpenPNM/blob/dev/examples/simulations/percolation/C_mixed_invasion_percolation.ipynb)
2. OpenPNM can NOT do viscous effects, it only has percolation algorithms at the moment, which assume quasistatic flow
3. This is how MixedInvasionPercolation works, so question 1 address this
4. This is also how MixedInvasionPercolation works, so also addressed by 1.

**Reply by livashu** [reply]
Thank you very much. I will refer to it.

**Reply by livashu** [reply]
Sorry to bother you.
Can't the pressure distribution be derived from the inlet and outlet pressure boundary conditions, as in stokes flow, and applied to the pressure of water intrusion?

**Reply by jgostick** [reply]
Not for 2-phase flow.  The movement of the capillary interfaces complicates things and we have not had the time or need to implement this.  The process is called 'dynamic invasion percolation' and there are numerous papers on it, including a few recent ones that use GPUs to speed up the solution, etc.

**Reply by livashu** [reply]
I understand.
Thank you for taking time out of your busy schedule to respond.
